src/mutation.cpp

```diff
@@ -891,6 +895,12 @@ void player::remove_mutation(std::string mut)
                 break;
             }
         }
+        for( size_t i = 0; i < mutation_data[traits_iter.first].prereqs2.size(); i++ ) {
+            if( mutation_data[traits_iter.first].prereqs2[i] == traits_iter.first ) {
+                dependant.push_back( traits_iter.first );
+                break;
+            }
+        }
```

Comment:
The whole "Check for dependant mutations first" makes no sense, it does not depend on the parameter `mut`, which means it does the same thing on every call. Why is it there?

And the check itself is nonsensical again: `mutation_data[traits_iter.first].prereqs[i] == traits_iter.first`
That reads "The prerequisite number `i` of mutation `traits_iter.first` is `traits_iter.first`", which means the mutation has itself as prerequisite.

And yes, this very old code, introduced with 16194ce2c3ef9c1b2220e7e6e9f6b2ff64748a09, aka PR 1523.